app.js

```diff
@@ -22,12 +34,106 @@ else {
     console.log('App starting with HTTP enabled');
 }
 
-app.get('/', function (req, res) {
-    res.send('Hello World');
+app.get('/', function (req, response) {
+    fs.readFile("views/html/login.html", function (err, data) {
+        if (err) throw err;
+        response.writeHead(200, { 'Content-Type': 'text/html' });
+        response.write(data);
+        response.end();
+    });
 });
 
+app.post('views/html/login', function (req, res) {
+    if (!req.hasOwnProperty('body') || req.body.hasOwnProperty('name'))
+        console.log("error logging in");
+
+    if (res.body.name == 'login-submit')
+        passport.authenticate('local', {
+            successRedirect: 'views/html/profile',
+            failureRedirect: '/views/html/login',
+            failureFlash: true
+        })
+    else {
+        storageModule.checkUser(email, 'local').then(function (data) {
+            if (Object.key(data).length == 0) {
+                storageModule.createUser(null, req.body.name, req.body.email, req.body.password, null, 'local').then(function (data) {
+                    res.redirect('views/html/profile');
+                }).otherwise(function (err) {
+                    res.redirect('/login');
+                });
+            }
+        }).otherwise(function (err) {
+            done(err, null);
+        });
+    }
+});
+
+passport.use(new LocalStrategy(
+    function (email, password, done) {
+        if (!email || !password)
+            return done({ 'err': { "msg": "Incomplete data", "code": "" } }, null);
+        storageModule.checkUser(email, 'local').then(function (data) {
+            if (Object.key(data).length == 0)
+                done(null, data);
+        }).otherwise(function (err) {
+            done(err, null);
+        });
+    }
+));
+
+passport.serializeUser(function (user, cb) {
+    cb(null, user);
+});
+
+passport.deserializeUser(function (obj, cb) {
+    cb(null, obj);
+});
+
+app.get('/auth/google', passport.authenticate('google', {
+    scope: ['email', 'profile']
+}
+));
+
+passport.use(new GoogleStrategy({
+    clientID: "594743792021-tvffn11n961cqea8mbufm9gnts372m0n.apps.googleusercontent.com",
```

Comment:
Make Client Id and Client Secret Configurable from configuration file with appropriate checks whether those fields are available or not. If not we will halt the application with some error. As this mechanism is crucial for the application.